tool/lib/model.dart

```diff
@@ -17,13 +17,24 @@ class DevToolsRepo {
   /// This returns the DevToolsRepo instance based on the current working
   /// directory.
   ///
-  /// This can fail and return null if the current working directory is not
-  /// contained within a git checkout of DevTools.
+  /// This will return null if the current working directory is not
+  /// contained within a git checkout of DevTools. Use [requireInstance] for a
+  /// descriptive exception in this case.
   static DevToolsRepo? getInstance() {
```

Comment:
will we ever have a use for this now? I wonder if it might be better to make this private with a `_` prefix, to prevent others from accidentally picking the wrong one in the future.

Reply:
Yeah, I noted this above - we could just remove it, or even put the `requireInstance` impl in here. I wasn't sure if there was a reason it was added as nullable though. If not, I think we just drop the `?` and make this behave like the new method below?

Reply:
I just replaced this instead. Now there's get `getInstance` (as before), but it's non-nullable and throws. As long as there's no code outside of this repo trying to use it, I think this is better.